spec/models/result_spec.rb

```diff
@@ -165,4 +165,41 @@
       end
     end
   end
+
+  describe '.set_release_on_results' do
+    let!(:assignment) { create :assignment_with_criteria_and_results }
+    let!(:released) { create :released_result }
+    it 'should raise StandardError with message not_complete error if result has not been completed' do
+      create(:incomplete_result)
+      begin
+        Result.set_release_on_results(assignment.groupings, true)
+      rescue StandardError => e
+        expect(e.message).to start_with(I18n.t('submissions.errors.not_complete'))
```

Comment:
You can check error messages using `raise_error`. See https://relishapp.com/rspec/rspec-expectations/v/2-8/docs/built-in-matchers/raise-error-matcher#match-message-with-a-string